opentech/static_src/src/app/src/redux/actions/notes.js

```diff
@@ -0,0 +1,46 @@
+import { updateSubmission } from '@actions/submissions';
+import api from '@api';
+
+export const FAIL_FETCHING_NOTES = 'FAIL_FETCHING_NOTES';
+export const START_FETCHING_NOTES = 'START_FETCHING_NOTES';
+export const UPDATE_NOTES = 'UPDATE_NOTES';
+export const UPDATE_NOTE = 'UPDATE_NOTE';
+
+const startFetchingNotes = () => ({
+    type: START_FETCHING_NOTES,
+});
+
+const failFetchingNotes = message => ({
+    type: FAIL_FETCHING_NOTES,
+    message,
+});
+
+export const fetchNotesForSubmission = submissionID => {
```

Comment:
We are starting to use this pattern quite a lot. Make an API request and handle `START`, `SUCCEED`, `FAIL`
I quite like what happens in the redux example. I might consider updating to this approach in the near future, I dont know if `redux-saga` or similar is required yet. What are your thoughts?

https://github.com/reduxjs/redux/blob/343cc97c194c69a1ac52e313efd6510e35a50703/examples/real-world/src/actions/index.js#L9-L15

https://github.com/reduxjs/redux/blob/343cc97c194c69a1ac52e313efd6510e35a50703/examples/real-world/src/middleware/api.js#L80